SonarQube.Client/Requests/RequestFactory.cs

```diff
@@ -94,17 +94,18 @@ public RequestFactory(ILogger logger)
         /// Creates a new TRequest implementation for the specified SonarQube version.
         /// </summary>
         /// <typeparam name="TRequest">The type of the request implementation to create.</typeparam>
-        /// <param name="version">
+        /// <param name="serverInfo">
         /// SonarQube version to return a request implementation for. When the provided value is null
         /// returns the registered implementation with the highest version number.
         /// </param>
         /// <returns>New the newest TRequest implementation for the specified SonarQube version.</returns>
-        public TRequest Create<TRequest>(Version version)
+        public TRequest Create<TRequest>(ServerInfo serverInfo)
             where TRequest : IRequest
         {
             SortedList<Version, Func<IRequest>> map;
             if (registrations.TryGetValue(typeof(TRequest), out map))
             {
+                var version = serverInfo?.Version;
```

Comment:
It seems weird that we would be doing a lookup with a null version. However, this is a special chicken-and-egg case that arises when initially connecting - at that point, we don't know what the version is because we haven't asked the server yet...